gslib/tests/test_rm.py

```diff
@@ -138,7 +138,11 @@ def _RunRemoveCommandAndCheck(self,
         stderr_set.remove('')  # Avoid groups represented by an empty string.
       if MACOS_WARNING in stderr_set:
         stderr_set.remove(MACOS_WARNING)
-      self.assertEqual(stderr_set, expected_stderr_lines)
+      if self._use_gcloud_storage:
```

Comment:
Yes, I mentioned the reason for removing condition in other PR. 
I am adding this validation separately because gcloud prints `rm` output/logs in more details compare to gsutil.
gcloud adds two more lines in the logs.
1. Removing buckets:
2. Removing objects:

So one way to validate this is to check that every stderrr line of gsutil output is present in gcloud storage output.

Reply:
Ok. Two follow-ups.
1. Why do we print extra information if the goal is to keep it exactly the same as gsutil ?
2. There seems to be a difference between the stderr output being compared against for gcloud and gsutil. It seems that for gcloud we do some sanitization.

Should we not do the same thing as the other PR for parity and rather assert only on the gsutil output, since that is what we expect for both cases after the formatter ?

Reply:
1. Well, We are not printing any thing extra here. It's just gcloud logs of command run.
2. We are not asserting gcloud output exactly same as gsutil because logs can be different and can be updated time to time. Here the main testing would be to check that every removal operation is the same as gsutil and happening with gcloud.

I don't think so because Formatters are only for formatting couple of boolean commands and the commands which print metadata in stdout. User should not rely on stderr output as it can change time by time. formatters are mostly for formatting metadata commands, I never saw a particular formatter is being used for formatting stderr, It would be different case if happens so.

Reply:
```
It's just gcloud logs of command run.
```
Sure, but why not truncate it ? Aren't we again assuming that anyone who depends on this output would have handled this case in their regex or whatever mechanism they might be using ?
If its difficult to equate, then we can take a call.

```
User should not rely on stderr output as it can change time by time
```
Why would it change ?

```
I never saw a particular formatter is being used for formatting stderr, It would be different case if happens so.
```
Perhaps they are originally always written for metadata, but I am not clear about why do we believe stderr like in this case should not be formatted ? Are we sure that all users of gsutil only rely on stdout (metadata info) and not stderr for their scripts ?

Reply:
Well, We are not going to update gsutil after shim only gsutil then gcloud storage command may print stderr lines of removing folders/managed folders but I don't think it will be supported gsutil. So gcloud logs can change in future. 
Regarding dependence on stdout only, It was written in the doc itself that shim will support stdout output translation because users are expected to parse stdout but not stderr.

Reply:
```
If its difficult to equate, then we can take a call.
```
I think we can come up with some new testing method for rm stderr messages. Though It will be part of adding coverage for parity tests as it will include considering various different scenarios that might arise in future.

Reply:
```

Reply:
I am saying these are the logs not the actual metadata output. We can not make sure that logs will be same forever in glcoud and user should not rely on them as well. Effectively I am checking whether every log line of gsutil's logs is present in shim logs or not. That way we can make sure that current gsutil behavior will not break but if some additional functionality comes in gcloud then gsutil doesn't care about that.

Reply:
Will discuss offline.

Reply:
@sushantkhanna can you please take a look!

Reply:
gcloud output would be,
```
$ gcloud storage rm -r gs://kk-test-bucket-13
Removing objects:
Removing gs://kk-test-bucket-13/file.txt#1718100087450900...                                                                
  Completed 1/1                                                                                                             
Removing buckets:
Removing gs://kk-test-bucket-13/...                                                                                         
  Completed 1/1                                
```

gsutil output would be,
```
$ gsutil rm -r gs://kk-test-bucket-13
Removing gs://kk-test-bucket-13/file.txt#1718099849459988...
/ [1 objects]                                                                   
Operation completed over 1 objects.                                              
Removing gs://kk-test-bucket-13/...
```
There are couple of unique log lines in both gsutil and gcloud  storage output.